test/cppsuite/test_harness/thread_context.h

```diff
@@ -0,0 +1,90 @@
+#ifndef THREAD_CONTEXT_H
+#define THREAD_CONTEXT_H
+
+#include <thread>
+
+#include "wiredtiger.h"
+
+namespace test_harness {
+enum class thread_operation { INSERT, UPDATE, READ, REMOVE, CHECKPOINT, TIMESTAMP, MONITOR };
+class thread_context {
+    public:
+    thread_context(
+      WT_SESSION *session, std::vector<std::string> collection_names, thread_operation type)
+        : _collection_names(collection_names), _running(false), _type(type)
+    {
+        _session = session;
+        _thread = nullptr;
+    }
+
+    thread_context(thread_operation type) : _running(false), _type(type)
+    {
+        _session = nullptr;
+        _thread = nullptr;
+    }
+
+    ~thread_context()
+    {
+        if (_session != nullptr)
+            testutil_die(DEBUG_ABORT,
+              "Session should've been cleaned up already. Did you forget to"
+              "call finish on the thread_manager?");
+        delete _thread;
+        _thread = nullptr;
+    }
+
+    /* Cleanup state. */
+    void
+    finish()
+    {
+        _running = false;
+        if (_thread != nullptr && _thread->joinable())
+            _thread->join();
+        if (_session != nullptr) {
+            if (_session->close(_session, NULL) != 0)
+                debug_info("Failed to close session for current thread", _trace_level, DEBUG_ERROR);
+            _session = nullptr;
+        }
+    }
+
+    const std::vector<std::string> &
+    get_collection_names() const
+    {
+        return _collection_names;
+    }
+
+    WT_SESSION *
+    get_session() const
+    {
+        return _session;
+    }
+
+    thread_operation
+    get_thread_operation() const
+    {
+        return _type;
+    }
+
+    bool
+    is_running() const
+    {
+        return _running;
+    }
+
+    void
+    set_thread(std::thread *thread)
```

Comment:
It is a bit weird that this also starts it.

Reply:
A thread is started when add_thread is called from thread_manager. Should we have _running = true elsewhere ?
it is true that we could call set_thread while the thread is not doing anything:
std::thread t;
set_thread(t)

Reply:
Can we have a separate function start()?

Reply:
Sure, I will create a sett for the running flag